Realm/Realm/Handles/SessionHandle.cs

```diff
@@ -209,22 +210,36 @@ public void UnsubscribeNotifications()
             }
         }
 
-        public async Task WaitAsync(ProgressDirection direction)
+        public async Task WaitAsync(ProgressDirection direction, CancellationToken? cancellationToken)
         {
             var tcs = new TaskCompletionSource();
+            if (cancellationToken?.IsCancellationRequested == true)
+            {
+                tcs.TrySetCanceled(cancellationToken.Value);
```

Comment:
I am not sure I understand why we need this and the following line. Can we just return?

Reply:
If cancellation is requested, the expected behavior would be to throw a `TaskCancelledException`, even if we didn't start doing the work. Regarding why we're cancelling the tcs vs throwing a `TaskCancelledException`, I guess both are roughly the same and don't have strong feelings about one or the other.